### ISSUE TITLE
performance issue

### ISSUE THREAD

**Issue by Jaware** [micrometer-metrics/micrometer#95]
performance test caused errors below：

```
java.lang.NullPointerException: null
        at java.util.LinkedList.get(LinkedList.java:477)
        at io.micrometer.core.instrument.stats.quantile.SlidingWindow.getAll(SlidingWindow.java:162)
        at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.getStreamSummary(WindowSketchQuantiles.java:245)
        at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.getFinalSummary(WindowSketchQuantiles.java:282)
        at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.get(WindowSketchQuantiles.java:100)
        at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$Child.collect(CustomPrometheusSummary.java:107)
```

**Comment by jkschneider** [comment]
I'm assuming you are referring to the `QuantilesBenchmark#windowQuantiles` benchmark. Can't seem to reproduce an NPE myself. Does the latest head still cause this issue for you?

**Comment by jkschneider** [comment]
Fixed in 1.0.0-rc.3 when Histogram/Quantile implementations were replaced by HdrHistogram.
